plugins/task-plugin/src/che-workspace-client.ts

```diff
@@ -109,7 +109,8 @@ export class CheWorkspaceClient {
     if (!machineExecServer) {
       throw new Error(`No server with type ${TERMINAL_SERVER_TYPE} found.`);
     }
-    return machineExecServer.url!;
+    const machineExecPort = machineExecServer.attributes!.port || '4444';
```

Comment:
I think it should not use hardcoded 4444 if the port is not there, just goes into error/invalid state

Reply:
if for some reasons it's not define in the workspace config, it is worth trying. At worst case, it would fail connecting.

Reply:
I can throw an error if you prefer, if port is undefined

Reply:
yes throw an error
as for example with devworkspaces attribute is targetPort and we may change the port value

Reply:
@benoitf done throwing an error